src/hotspot/share/runtime/stackOverflow.cpp

```diff
@@ -103,10 +103,7 @@ void StackOverflow::create_stack_guard_pages() {
   } else {
     log_warning(os, thread)("Attempt to protect stack guard pages failed ("
       PTR_FORMAT "-" PTR_FORMAT ").", p2i(low_addr), p2i(low_addr + len));
-    if (os::uncommit_memory((char *) low_addr, len)) {
-      log_warning(os, thread)("Attempt to deallocate stack guard pages failed.");
-    }
-    return;
```

Comment:
I always was curious about this uncommit. I wondered whether it was a fallback for platforms where mprotect would not work on thread stacks. But then, why warn unconditionally?

Reply:
Maybe the uncommit was an attempt to achieve the same thing as protecting the pages, ie get a SEGV when accessing them. I don't know either.